client/wrappers/retryable/admin.go

```diff
@@ -0,0 +1,366 @@
+// The MIT License (MIT)
+
+// Copyright (c) 2017-2020 Uber Technologies Inc.
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+package retryable
+
+// Code generated by gowrap. DO NOT EDIT.
+// template: ../../templates/retry.tmpl
+// gowrap: http://github.com/hexdigest/gowrap
+
+import (
+	"context"
+
+	"go.uber.org/yarpc"
+
+	"github.com/uber/cadence/client/admin"
+	"github.com/uber/cadence/common/backoff"
+	"github.com/uber/cadence/common/types"
+)
+
+// adminClient implements admin.Client interface instrumented with retries
+type adminClient struct {
+	client        admin.Client
+	throttleRetry *backoff.ThrottleRetry
+}
+
+// NewAdminClient creates a new instance of adminClient with retry policy
+func NewAdminClient(client admin.Client, policy backoff.RetryPolicy, isRetryable backoff.IsRetryable) admin.Client {
+	return &adminClient{
+		client: client,
+		throttleRetry: backoff.NewThrottleRetry(
+			backoff.WithRetryPolicy(policy),
+			backoff.WithRetryableError(isRetryable),
+		),
+	}
+}
+
+func (c *adminClient) AddSearchAttribute(ctx context.Context, ap1 *types.AddSearchAttributeRequest, p1 ...yarpc.CallOption) (err error) {
+	op := func() error {
+		return c.client.AddSearchAttribute(ctx, ap1, p1...)
+	}
+	return c.throttleRetry.Do(ctx, op)
+}
+
+func (c *adminClient) CloseShard(ctx context.Context, cp1 *types.CloseShardRequest, p1 ...yarpc.CallOption) (err error) {
+	op := func() error {
+		return c.client.CloseShard(ctx, cp1, p1...)
+	}
+	return c.throttleRetry.Do(ctx, op)
+}
+
+func (c *adminClient) CountDLQMessages(ctx context.Context, cp1 *types.CountDLQMessagesRequest, p1 ...yarpc.CallOption) (cp2 *types.CountDLQMessagesResponse, err error) {
+	var resp *types.CountDLQMessagesResponse
+	op := func() error {
+		var err error
+		resp, err = c.client.CountDLQMessages(ctx, cp1, p1...)
+		return err
+	}
+	err = c.throttleRetry.Do(ctx, op)
+	return resp, err
```

Comment:
entirely unrelated to this PR and not worth changing here, but we do have generics now.... this could be pretty simply reduced to
```go
return c.throttleRetry.Do1(ctx, c.client.CountDLQMessages, cp1, p1...)
// and the void-return ones would be
return c.ThrottleRetry.Do(ctx, c.client.CloseShard, cp1, p1...)
```